### PR TITLE
Health indicators

### PR DESCRIPTION
**Pull request by dcarver1**
Lots of new functions in here. 

Big picture 

-- Health indicators 
--- cancer, diabetes, heart 
---- please look at logic behind bring these values together. We're combining county level data from CDC places with tract level mortality data from the state. 
Attached a new mortality dataset with CSV version of a xls files. 
[mortalityData.zip](https://github.com/user-attachments/files/17217683/mortalityData.zip)

ACS data
- gatherACS function is the tidy census download 
-- indicator function take this input and select out the specific measures of interest. 
-- really very standardize so there shouldn't be a lot of variability between functions 

For specific functions checking for the clean run will be great

### CODE REVIEW

File: functions/indicatorScoreCalculation/over65.R [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# name <- names(geometryFiles)[[1]]
+
+processOver65 <- function(geometry, name, data){
+  # select the data set of interest 
+  vals <- data[[grep(pattern = name, x = names(data))]] |> as.data.frame()
+  
+  # structure then generate and select measures of concern
+  output <- structureACS(vals) |>
+    dplyr::mutate(
+      age_over65 = sum(B01001_020, B01001_021))|>
```
Comment (ccmothes) [inline functions/indicatorScoreCalculation/over65.R:13]: Double-checking this - there are only two variables being summed here while 12 different variables were downloaded from the ACS for over 65.

File: functions/indicatorScoreCalculation/lowIncome.R [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# name <- names(geometryFiles)[[1]]
+
+processLowIncome <- function(geometry, name, data){
+  # select the data set of interest 
+  vals <- data[[grep(pattern = name, x = names(data))]] |> as.data.frame()
+  
+  # structure then generate and select measures of concern
+  output <- structureACS(vals) |>
+    dplyr::mutate(
+      percent_lowincome = ifelse(C17002_001 == 0, NA, (C17002_001 - C17002_008) / C17002_001))|>
```
Comment (ccmothes) [inline functions/indicatorScoreCalculation/lowIncome.R:13]: Curious what variable C17002_008 represents?

**Review by ccmothes — APPROVED** [review]
A couple general comments/questions

- I couldn't run the EJscreen-related functions without the data, but I noticed lot of repetitive code for all of these functions. I think you could condense that into a single function that processes all EJ vars at once?
- Some clarification on the logic for the health indicators...For each once, the mortality value is always the same, correct? I'm worried that the output variables for each one that become e.g., cancerMortality_pcntl, heartMortality_pcntl is a bit misleading since they are the same morality values that are not directly tied to each indicator. 

Other comments added to in-line code.

File: 0_main.R [hunk 3]
```diff
@@ -57,42 +57,121 @@ vals <- readRDS("data/processed/geographies/bgNeighbors.RDS")
...
 # environmental Effects ---- 
+## impaired streams 
+
+## hazardous waste 
+
+## mining
 getMining(geometryLayers = geometries)
 
+## NPL sites 
+getNPSsites(geometryLayers = geometries)
```
[the remark this answers is not in the document]
Comment (dcarver1) [reply 0_main.R:114]: yes good catch

File: utilities/structureACS.R [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# helper function to transforming the long to wide data structure of ACS data
+structureACS <- function(data){
+  data |>
+    tidyr::spread(key = variable, value = estimate) |>
+    dplyr::group_by(GEOID) |>
+    dplyr::summarize(across(contains("_"), ~ sum(.x, na.rm = TRUE))) |>
+    dplyr::group_by(GEOID)
```
[the remark this answers is not in the document]
Comment (dcarver1) [reply utilities/structureACS.R:7]: This output passes into a mutate which requires the group by call. It is confusing as written so I've pulled it from the function and added it to the specific processing functions

File: functions/indicatorScoreCalculation/lowIncome.R (hunk 2 above)
Comment (dcarver1) [reply functions/indicatorScoreCalculation/lowIncome.R:13]: There is a export of the ACS label values in data/products/acsLabels.csv

The ACS code was provide by CDPHE during the development of 1.0, so I haven't changed it much. 

the census table is ratio of income to poverty level in the past 12 months
001 is the total pop 
008 is the pop with an income of 2x poverty level

File: functions/indicatorScoreCalculation/over65.R (hunk 1 above)
Comment (dcarver1) [reply functions/indicatorScoreCalculation/over65.R:13]: thanks for the catch, 
included the range of indicators for both male 20:25 and female 44:49

**Comment by dcarver1** [comment]
_I couldn't run the EJscreen-related functions without the data, but I noticed lot of repetitive code for all of these functions. I think you could condense that into a single function that processes all EJ vars at once?_
    - we processed the ejscreen variable in 1.0 as a single function. Decided to move to this structure so that every indicator has a specific processing function. They removes the need of the user to understand what indicators are from the ejscreen dataset 
    
    
    Some clarification on the logic for the health indicators...For each once, the mortality value is always the same, correct? I'm worried that the output variables for each one that become e.g., cancerMortality_pcntl, heartMortality_pcntl is a bit misleading since they are the same morality values that are not directly tied to each indicator.
- If you could clarify on this point that would be appreciated. 
- The heath indicator combines a county level age adjusted prevalence dataset with a census tract level mortality datasets, because never of these indicators are measured at the block group the combined measure at the block group will be the same as the census track that it is contained within in. This does result in quite a bit of duplication in score/rank.

**Comment by dcarver1** [comment]
Ejscreen data can be gathered from the link below 

https://www.epa.gov/ejscreen/download-ejscreen-data 

using the national level csv 
image

**Pull request merged**